DE-image-featurizer/evaluate/evaluator.py

```diff
@@ -0,0 +1,380 @@
+            self.train_ds = tf.keras.utils.image_dataset_from_directory(data_dir + '/train',
+                                                    seed=SEED, batch_size=self.batch_size)
+            self.val_ds = tf.keras.utils.image_dataset_from_directory(data_dir + '/test',
+                                                    seed=SEED, batch_size=self.batch_size)
+        else:
+            self.train_ds = tf.keras.utils.image_dataset_from_directory(data_dir,
+                                                    validation_split=self.validation_split,
+                                                    subset='training',
+                                                    seed=SEED, batch_size=self.batch_size)
+            self.val_ds = tf.keras.utils.image_dataset_from_directory(data_dir,
+                                                    validation_split=self.validation_split,
+                                                    subset='validation',
+                                                    seed=SEED, batch_size=self.batch_size)
+
+    def _load_featurizer(self, featurizer):
+        """
+        Instantiates the model from TF-hub library or
+        from the featurizers present in the DE-image-featurizer repository
+        and also creates a preprocessor to standardise the input.
+
+        Parameters
+        -------------
+        featurizer: string
+            Name of a traditional CV featurizer, or URL to a tf-hub featurizer model.
+            Accepted values for names are 'hog', 'lbp', 'orb' and 'sift'.
+        """
+        if featurizer.lower() in FEATURIZER_DICT:
+            self.featurizer = FEATURIZER_DICT[featurizer.lower()].fit_transform
+        else:
+            self.featurizer = hub.KerasLayer(featurizer,
+                                             input_shape=(224, 224, 3), trainable=False)
+        self.preprocessor = tf.keras.Sequential([tf.keras.layers.Resizing(224, 224),
+                                                 tf.keras.layers.Rescaling(1. / 255)])
+
+    def _extract_features(self, is_train):
+        """
+        Computes the features for the input data using the image featurizer.
+
+        Parameters
+        ----------
+        is_train: boolean
+            To distinguish between training and testing set
+
+        Returns
+        -------
+        features_array: numpy.ndarray of shape (train_data_size,num_of_features)
+            Numpy array containing extracted features
+
+        input_labels: numpy.ndarray of shape (train_data_size,)
+            Numpy array containing corresponding labels
+
+        """
+        features_array = None
+        input_labels = None
+        input_data = self.train_ds if is_train else self.val_ds
+        for image_batch, labels_batch in tqdm(input_data):
+            preprocessed_image_batch = self.preprocessor(image_batch)
+            image_batch_features = self.featurizer(preprocessed_image_batch)
+            if features_array is None:
```

Comment:
@rahulthakurakridata Try avoiding multiple if-else statements. Maybe, you can append all features to a list and do `np.asarray` over it.